### PR TITLE
Fix router

### LINKED ISSUE #448: Restructure Router
### Overview
Currently, HeadNav and FooterNav are rendered on every page because in frontend/src/App.tsx, the App returns: 
```
    <>
      <HeaderNav />
      <RouterProvider router={router} />
      <FooterNav />
    </>
```

Login and Signup page do not need the Footer, and do not share the same HeaderNav with the other page, so we should restructure the routes.

### Action Items

- [ ] Restructure RouterProvider in frontend/src/App.tsx 
- [ ] Fix frontend/src/router/Router.tsx 

### Resources/Instructions
[react-router-dom nested routes](https://reactrouter.com/en/main/start/overview#nested-routes)

### PR DESCRIPTION
**Pull request by bzzz-coding**
Fixes #448 

### Changes

- Restructured router  
  - Nested Landing, QualifierPage, CreditsPage, Demo, Demo-Tailwind, and NotFoundPage that need both default header nav and footer nav under DefaultNavLayout
  - Nested DefaultNavLayout, login and signup under HomeLayout
- Removed header and footer components from the App.tsx return
- Added Link to login button in default header nav and removed href
- Moved SimpleNav out of HeaderNav component and added to AuthenticationPage
- Added `"./src/components/Navigation/SimpleNav.tsx"` to tailwind config content

### CODE REVIEW

File: frontend/src/components/Navigation/HeaderNav.tsx [hunk 1]
```diff
@@ -28,8 +28,6 @@ const menuItems: menuObject[] = [
  */
```
Comment (MattPereira) [inline frontend/src/components/Navigation/HeaderNav.tsx:28]: we can delete the comments just above this line since you fixed the issue they mention :tada:

**Review by MattPereira — APPROVED** [review]
@bzzz-coding magnificent refactor!!! :sparkles: 

Love the updated syntax in the `Router.tsx` file and the `HeaderNav.tsx` is sooooo much cleaner :raised_hands: Really appreciate you schooling me on react router best practices :sweat_smile: 

Just a few ideas to consider:
- Maybe `AuthNav` would be a better name than `SimpleNav` since its specific to the auth page?
- I created the `tw-components` folder with intention of trying to separate components that were styled with tailwind from components styled with sass. I forgot to add `HeaderNav` after converting it to tw. Do you like the idea or do you prefer to mix the tw components with sass components in the regular `components` folder?
- Maybe `Layouts/` folder should go at same level as `router/` and `pages/` ? This might just be my bias as i've grown accustomed to nextjs organization patterns but I do think its kinda weird to keep `api_data` folder inside of `pages/` directory

File: frontend/src/components/Navigation/HeaderNav.tsx (hunk 1 above)
Comment (bzzz-coding) [reply frontend/src/components/Navigation/HeaderNav.tsx:28]: Ah you're right, yay!

**Comment by bzzz-coding** [comment]
> @bzzz-coding magnificent refactor!!! ✨
> 
> Love the updated syntax in the `Router.tsx` file and the `HeaderNav.tsx` is sooooo much cleaner 🙌 Really appreciate you schooling me on react router best practices 😅
> 
> Just a few ideas to consider:
> 
> * I created the `tw-components` folder with intention of trying to separate components that were styled with tailwind from components styled with sass. I forgot to add `HeaderNav` after converting it to tw. This suggestion also comes with benefit of not having to add as many individual files to the `content:` of `tailwind.config.js`
> * Maybe `Layouts/` folder should go at same level as `router/` and `pages/` or inside of `router/` ? This might just be my bias as i've grown accustomed to nextjs organization patterns but I do think its kinda weird to keep `api_data/` folder inside of `pages/` directory


Aww thanks @MattPereira for reviewing my PR and for your suggestions! 
- Got it! I'll move the HeaderNav to tw-components and delete that single file from tailwind config contents. I agree, this would help prevent the tailwind config contents getting out of control as we add more components in the future. 
- Makes sense! I'll work on restructuring the folders. Go Next.js! 🚀

**Review by MattPereira — APPROVED** [review]
chefs-kiss-french-chef

**Pull request merged**